File: UnassignedTicket.OutlookAddIn/Models/Ticket.cs
using System;

namespace UnassignedTicket.OutlookAddIn.Models
{
    internal sealed class Ticket
    {
        public string Id { get; set; }
        public string Type { get; set; }
        public string Summary { get; set; }
        public string Priority { get; set; }
        public string Status { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? GroupAssignedAt { get; set; }
        public string Url { get; set; }

        public TimeSpan WaitingTime
        {
            get { return DateTime.Now - (GroupAssignedAt ?? CreatedAt); }
        }
    }
}

File: src/types/ticket.ts
export type TicketType = "INC" | "WO";

export interface Ticket {
  id: string;
  type: TicketType;
  summary: string;
  priority: string;
  status: string;
  createdAt: string;
  groupAssignedAt: string | null;
  url: string | null;
}

export interface TicketCounts {
  INC: number;
  WO: number;
}

export interface UnassignedTicketsResponse {
  group: string;
  generatedAt: string;
  total: number;
  counts: TicketCounts;
  tickets: Ticket[];
}

export type TicketFilter = "ALL" | TicketType;

